docs/zkEmailDocs/UsageGuide/README.md

```diff
@@ -106,9 +106,9 @@ To compile the circuit locally, you need to have Rust and Circom installed first
 
 
 ```bash
-circom MyCircuit.circom -o --r1cs --wasm --sym --c 
+circom MyCircuit.circom -o --r1cs --wasm --sym --c --O0
 ```
-*Note: You can add -l to specify the directory where the directive `include` should look for the circuits indicated. For our repo use circom -l node_modules instead of circom.
+*Note: You can add -l to specify the directory where the directive `include` should look for the circuits indicated. For our repo use `circom -l node_modules` instead of circom. Additionally, we generally recommend using the `--O0` flag for optimization during compilation for beginners. However, if you're more experienced with Circom, feel free to use the `--O1` flag instead. It's important to avoid using the `--O2` flag as that is the default setting and it may lead to the deletion of additional constraints.*  
```

Comment:
I think using 02 flag (default) is ok. 
@Divide-By-0  Addition constraints are removed during optimization by subtracting the added value from `out` instead..right? 
`out = a * b + c` become `out  - c = a * b`, but the addition is still part of the R1CS?

Reply:
No. Addition is removed from the r1cs. See https://stackoverflow.com/questions/78136647/circom-does-not-create-a-constraint-for-addition/78177349#78177349

And

https://stackoverflow.com/questions/77688466/circom-compiler-removes-crucial-constraint-after-simplication/78177354?noredirect=1#comment137833229_78177354

@Metachaser24 can you add these links as references to the docs?

Reply:
yes adding now

Reply:
Did a quick test:
```
signal d <== b + 5;
c === a * d;
```

With -O0
```
[  ] * [  ] - [ 51 +main.b +21888242871839275222246405745257275088548364400416034343698204186575808495616main.d ] = 0
[ 21888242871839275222246405745257275088548364400416034343698204186575808495616main.a ] * [ main.d ] - [ 21888242871839275222246405745257275088548364400416034343698204186575808495616main.c ] = 0
```

With -O2
```
[ 21888242871839275222246405745257275088548364400416034343698204186575808495616main.a ] * [ 51 +main.b ] - [ 21888242871839275222246405745257275088548364400416034343698204186575808495616main.c ] = 0
```

Even though addition constraint is removed in O2, the result is effectively same. The addition output gets added on to the next constraint where its used. 

Even when circuit has only one addition contract we cannot generate witness that doesn't match the condition even though r1cs are empty (maybe because of the wire config in .sym file)

Reply:
@Divide-By-0 @Metachaser24 I think O2 (default) should be used ideally as it reduce the total constraints without loosing intended assertions. 
Or am I missing something?

Reply:
> @Divide-By-0 @Metachaser24 I think O2 (default) should be used ideally as it reduce the total constraints without loosing intended assertions. Or am I missing something?

I think there are cases where they do corrupt the logic of the circuit. One example is an addition constraint unconnected to anything else -- we used to have this for address plus one which wasn't constrained -- Pierre made a fake input pass for a similar case. I'm worried the pattern appears for other combinations of + and * as well.

Reply:
In the worst case, it also gives proof malleability -- i.e. a nullifier constraint might have an addition somewhere then people might be able to forge a different nullifier that technically proves knowledge of preimage, but may not satisfy uniqueness any longer (since that specific value is no longer constrained).